dpe/src/commands/derive_child.rs

```diff
@@ -373,6 +378,80 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_full_attestation_flow() {
+        let mut env = DpeEnv::<TestTypes> {
+            crypto: OpensslCrypto::new(),
+            platform: DefaultPlatform,
+        };
+        let mut dpe = DpeInstance::new(
+            &mut env,
+            Support::AUTO_INIT | Support::INTERNAL_INFO | Support::X509,
+        )
+        .unwrap();
+
+        DeriveChildCmd {
+            handle: ContextHandle::default(),
+            data: [0; DPE_PROFILE.get_tci_size()],
+            flags: DeriveChildFlags::RETAIN_PARENT,
+            tci_type: 0,
+            target_locality: 0,
+        }
+        .execute(&mut dpe, &mut env, TEST_LOCALITIES[0])
+        .unwrap();
+
+        let sig = {
+            let cmd = SignCmd {
+                handle: ContextHandle::default(),
+                label: TEST_LABEL,
+                flags: SignFlags::empty(),
+                digest: TEST_DIGEST,
+            };
+            let resp = match cmd.execute(&mut dpe, &mut env, TEST_LOCALITIES[0]).unwrap() {
+                Response::Sign(resp) => resp,
+                _ => panic!("Incorrect response type"),
+            };
+
+            EcdsaSig::from_private_components(
+                BigNum::from_slice(&resp.sig_r_or_hmac).unwrap(),
+                BigNum::from_slice(&resp.sig_s).unwrap(),
+            )
+            .unwrap()
+        };
+
+        // necessary so that generate_new_handle doesn't fail due to non-deterministic random bytes
+        dpe.contexts[1].handle = ContextHandle([0x2; ContextHandle::SIZE]);
```

Comment:
I don't understand what this is for. If you return the output handle from each of the responses, shouldn't it just work to use the handle that DPE generated?

Reply:
When we call derive_child a second time without the make_default flag, it will call generate_new_handle which will generate another buffer of 1s (the first buffer of 1s was generated in the first call to derive_child and is located at contexts[1].) We can't have duplicate buffers, so I did this to make generate_new_handle work for the second derive_child call. For non-deterministic rand_bytes, this isn't an issue, and we wouldn't have had to do this. 

As for the test overall, I am trying to derive new contexts with contexts[0] as the parent, and then just pass the default context to certify key and sign to check that internal_input_info isn't getting included unecessarily. This is why using the output handle isn't that helpful.

Reply:
oh, because the RNG is fake. Maybe we should hash or increment the previous value every time for the test RNG. This will give us better testing properties, but will still be deterministic.

Reply:
We already do increment the previous value. The issue is the buffer passed to generate_new_handle is always the buffer of 0s, so we still get a buffer of 1s. Even if we hashed it, we would get the same problem

Reply:
Actually maybe a better option: Use the rng crate and initialize it `from_seed` (https://docs.rs/rand/latest/rand/trait.SeedableRng.html#tymethod.from_seed). This will be deterministic but still give reasonable RNG properties if called more than once.

Reply:
For clarity, I mean call Rng::from_seed when creating OpensslCrypto and saving it as a member (since calling it in `rand_bytes` would lead to the same problem)

Reply:
Done.